src/CommaSeparatedParser.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace MPScholten\RequestParser;
+
+class CommaSeparatedParser extends AbstractValueParser
+{
+    const INT = 'INT';
+    const STRING = 'STRING';
+    const FLOAT = 'FLOAT';
+    const BOOLEAN = 'BOOLEAN';
+    const YES_NO_BOOLEAN = 'YES_NO_BOOLEAN';
+    const JSON = 'JSON';
+    const DATE_TIME = 'DATE_TIME';
+
+    private $type;
+
+    protected function parse($value)
+    {
+        if (!isset($this->type)) {
+            $this->type = self::STRING;
+        }
+        $valuesArray = explode(",", $value);
+        switch ($this->type) {
```

Comment:
This approach has a disadvantage: it will break autocompletion (atleast a bit).

e.g. `$this->queryParameter(...)->commaSeparated()->int()->required()` will not be typed as `int[]`. It will just be `array`. And IMO this is bad.

For example inside a foreach the `int[]` type would help the IDE to catch errors. E.g.:
```php
$ints = $this->bodyParameter('ints')->commaSeperated()->int()->required();
foreach ($ints as $int) {
    doSomething($int);
}
function doSomething(SomeObjectNotAnInteger $a) {};
```

Here the IDE cannot catch that `doSomething($int)` is a type error.